### PR TITLE
CASM-4231 Enhanced Logging for IUF Scripts

### PR DESCRIPTION
**Pull request by rajeshranganathan-hpe**
## Summary and Scope

This change is to support enhanced logging for scripts used by IUF operations. The "argo_entrypoint.sh" script is used by the "vcs-upload" operation of the "deliver-product" stage of IUF. The changes introduce log levels to each statement that logs information for user. 

See the following link for the feedback which prompted the changes:
https://rndwiki-pro.its.hpecorp.net/display/~john.navitsky@hpe.com/IUF+Stage+Output

## Issues and Related PRs
None

## Testing

Planned for test in #gamora.

### Tested on:

  * `gamora`

### Test description:

Run the "deliver-product" stage for the "COS" product to verify.

## Risks and Mitigations

None

## Pull Request Checklist

- [x] Target branch correct
- [X] Testing is appropriate and complete, if applicable

### CODE REVIEW

File: argo_entrypoint.sh [hunk 1]
```diff
@@ -37,14 +37,16 @@ update-ca-certificates --fresh 2>/dev/null
 
 until curl --head ${CF_IMPORT_GITEA_URL}; \
 do
-  echo Waiting for Gitea API to be available; \
+  echo "INFO Waiting for 3 seconds for Gitea API to be available"; \
   sleep 3; \
 done; \
-echo Gitea API available;
+echo "INFO Gitea API available";
```
Comment (JohnNavitskyHPE) [inline argo_entrypoint.sh:43]: debug

File: argo_entrypoint.sh [hunk 2]
```diff
@@ -37,14 +37,16 @@ update-ca-certificates --fresh 2>/dev/null
 
 until curl --head ${CF_IMPORT_GITEA_URL}; \
 do
-  echo Waiting for Gitea API to be available; \
+  echo "INFO Waiting for 3 seconds for Gitea API to be available"; \
```
Comment (JohnNavitskyHPE) [inline argo_entrypoint.sh:40]: debug

**Review by JohnNavitskyHPE — APPROVED** [review]

File: argo_entrypoint.sh [hunk 3]
```diff
@@ -37,14 +37,16 @@ update-ca-certificates --fresh 2>/dev/null
 
 until curl --head ${CF_IMPORT_GITEA_URL}; \
 do
-  echo Waiting for Gitea API to be available; \
+  echo "DEBUG Waiting for 3 seconds for Gitea API to be available"; \
```
Comment (mharding-hpe) [inline argo_entrypoint.sh:40]:
```suggestion
  echo "DEBUG: Waiting for 3 seconds for Gitea API to be available"; \
```

File: argo_entrypoint.sh [hunk 4]
```diff
@@ -37,14 +37,16 @@ update-ca-certificates --fresh 2>/dev/null
...
   sleep 3; \
 done; \
-echo Gitea API available;
+echo "DEBUG Gitea API available";
```
Comment (mharding-hpe) [inline argo_entrypoint.sh:43]:
```suggestion
echo "DEBUG: Gitea API available";
```

File: argo_entrypoint.sh [hunk 5]
```diff
@@ -37,14 +37,16 @@ update-ca-certificates --fresh 2>/dev/null
...
 
 # Overwrite content before import
-cp -r "/shared/"* ${CF_IMPORT_CONTENT}/ && echo "overwriting /content success" || echo "overwriting /content failed"
+cp -r "/shared/"* ${CF_IMPORT_CONTENT}/ && echo "INFO overwriting /content success" || echo "ERROR overwriting /content failed"
```
Comment (mharding-hpe) [inline argo_entrypoint.sh:46]:
```suggestion
cp -rv "/shared/"* "${CF_IMPORT_CONTENT}/" && echo "INFO: overwriting /content success" || echo "ERROR: overwriting /content failed"
```

File: argo_entrypoint.sh [hunk 6]
```diff
@@ -37,14 +37,16 @@ update-ca-certificates --fresh 2>/dev/null
...
 
 # Import the configuration content
 cd /opt/csm/cf-gitea-import
+echo "DEBUG Starting to import the contents of the ${CF_IMPORT_CONTENT} directory into Gitea"
```
Comment (mharding-hpe) [inline argo_entrypoint.sh:50]:
```suggestion
echo "DEBUG: Starting to import the contents of the ${CF_IMPORT_CONTENT} directory into Gitea"
```

**Review by mharding-hpe — CHANGES_REQUESTED** [review]
Why are some of the logging levels in the Python script changed from info to debug? That means they will show up less often. Is that desired? The PR sounds like it is trying to increase the amount of logging, but that would decrease it. Is that intended?

**Review by mharding-hpe — APPROVED** [review]

**Comment by JohnNavitskyHPE** [comment]
> Why are some of the logging levels in the Python script changed from info to debug? That means they will show up less often. Is that desired? The PR sounds like it is trying to increase the amount of logging, but that would decrease it. Is that intended?

so the overall tone that was attempted (across all the operations) were things like:
* don't tell the user you are about to do something, then tell them you did it if those operations are in close succession.  move the "i am about to do something" into debug since it may be useful if there is a problem since it helps identify which area of code you're in, but instead, focus on printing an info or error about the RESULT of that operation.  
* move stuff that's basically internal housekeeping stuff to debug.   so for example, don't tell them you checked out a copy of the repo, just tell them that you imported branch xyz or you merged branch foo into bar.  since we moved the other messages to debug, if there's an actual problem they need to debug, the information is still there but the flow of the output to the command line is much easier to follow.

**Pull request merged**